File: pidev3A19/src/main/java/models/Compagne.java
package models;

import java.sql.Date;

public class Compagne {
    private String nom_sponsor;
    private int id_compagne;
    private Date date_debut;
    private Date date_fin;
    private String logo_compagne;
    private String TypeMarketing;
    private String status;
    private float tarifs;
    private Produit produit;

    // Constructors
    public Compagne() {
        this.produit = new Produit();
    }

    public Compagne(String nom_sponsor, int id_compagne, Date date_debut, Date date_fin, String logo_compagne, String TypeMarketing, String status, float tarifs) {
        this.nom_sponsor = nom_sponsor;
        this.id_compagne = id_compagne;
        this.date_debut = date_debut;
        this.date_fin = date_fin;
        this.logo_compagne = logo_compagne;
        this.TypeMarketing = TypeMarketing;
        this.status = status;
        this.tarifs = tarifs;
        this.produit = new Produit();
    }

    public Compagne(String nom_sponsor, String date_debut, String date_fin, String logo_compagne, String typeMarketing, String status, float tarifs, Produit produit) {
        this.nom_sponsor = nom_sponsor;
        this.date_debut = Date.valueOf(date_debut);
        this.date_fin = Date.valueOf(date_fin);
        this.logo_compagne = logo_compagne;
        this.TypeMarketing = typeMarketing;
        this.status = status;
        this.tarifs = tarifs;
        this.produit = produit;
    }

    public Compagne(String riovaciar, Date date, Date date1, String image, String digital, String active, double v, int idProduit) {
    }

    // Getters and setters
    public String getNom_sponsor() {
        return nom_sponsor;
    }

    public void setNom_sponsor(String nom_sponsor) {
        this.nom_sponsor = nom_sponsor;
    }

    public int getId_compagne() {
        return id_compagne;
    }

    public void setId_compagne(int id_compagne) {
        this.id_compagne = id_compagne;
    }

    public Date getDate_debut() {
        return date_debut;
    }

    public void setDate_debut(Date date_debut) {
        this.date_debut = date_debut;
    }

    public Date getDate_fin() {
        return date_fin;
    }

    public void setDate_fin(Date date_fin) {
        this.date_fin = date_fin;
    }

    public String getLogo_compagne() {
        return logo_compagne;
    }

    public void setLogo_compagne(String logo_compagne) {
        this.logo_compagne = logo_compagne;
    }

    public String getTypeMarketing() {
        return TypeMarketing;
    }

    public void setTypeMarketing(String typeMarketing) {
        this.TypeMarketing = typeMarketing;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public float getTarifs() {
        return tarifs;
    }

    public void setTarifs(float tarifs) {
        this.tarifs = tarifs;
    }

    public Produit getProduit() {
        return produit;
    }

    public void setProduit(Produit produit) {
        this.produit = produit;
    }

    @Override
    public String toString() {
        return "Compagne{" +
                "id_compagne=" + id_compagne +
                ", nom_sponsor='" + nom_sponsor + '\'' +
                ", date_debut=" + date_debut +
                ", date_fin=" + date_fin +
                ", logo_compagne='" + logo_compagne + '\'' +
                ", TypeMarketing='" + TypeMarketing + '\'' +
                ", status='" + status + '\'' +
                ", tarifs=" + tarifs +
                ", produit=" + produit +
                '}';
    }
}

File: src/Entity/Compagne.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use App\Repository\CompagneRepository;

#[ORM\Entity(repositoryClass: CompagneRepository::class)]
#[ORM\Table(name: 'compagne')]
class Compagne
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column(type: 'integer')]
    private ?int $id = null;

    #[ORM\Column(type: 'string', length: 255, nullable: false)]
    private string $nomSponsor;

    #[ORM\Column(type: 'float', nullable: false)]
    private float $tarifs;

    #[ORM\Column(type: 'date', nullable: false)]
    private \DateTimeInterface $dateDebut;

    #[ORM\Column(type: 'date', nullable: false)]
    private \DateTimeInterface $dateFin;

    #[ORM\Column(type: 'string', length: 255, nullable: false)]
    private string $logoCompagne;

    #[ORM\Column(type: 'string', length: 50, nullable: false)]
    private string $status;

    #[ORM\Column(type: 'string', length: 100, nullable: false)]
    private string $typeMarketing;

    #[ORM\ManyToOne(targetEntity: Produit::class, inversedBy: 'compagnes')]
    #[ORM\JoinColumn(
        name: 'id_produit', 
        referencedColumnName: 'id_produit',
        nullable: false,
        onDelete: 'RESTRICT'
    )]
    private Produit $produit;

    // Getters et Setters

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getNomSponsor(): string
    {
        return $this->nomSponsor;
    }

    public function setNomSponsor(string $nomSponsor): self
    {
        $this->nomSponsor = $nomSponsor;
        return $this;
    }

    public function getTarifs(): float
    {
        return $this->tarifs;
    }

    public function setTarifs(float $tarifs): self
    {
        $this->tarifs = $tarifs;
        return $this;
    }

    public function getDateDebut(): \DateTimeInterface
    {
        return $this->dateDebut;
    }

    public function setDateDebut(\DateTimeInterface $dateDebut): self
    {
        $this->dateDebut = $dateDebut;
        return $this;
    }

    public function getDateFin(): \DateTimeInterface
    {
        return $this->dateFin;
    }

    public function setDateFin(\DateTimeInterface $dateFin): self
    {
        $this->dateFin = $dateFin;
        return $this;
    }

    public function getLogoCompagne(): string
    {
        return $this->logoCompagne;
    }

    public function setLogoCompagne(string $logoCompagne): self
    {
        $this->logoCompagne = $logoCompagne;
        return $this;
    }

    public function getStatus(): string
    {
        return $this->status;
    }

    public function setStatus(string $status): self
    {
        $this->status = $status;
        return $this;
    }

    public function getTypeMarketing(): string
    {
        return $this->typeMarketing;
    }

    public function setTypeMarketing(string $typeMarketing): self
    {
        $this->typeMarketing = $typeMarketing;
        return $this;
    }

    public function getProduit(): Produit
    {
        return $this->produit;
    }

    public function setProduit(Produit $produit): self
    {
        $this->produit = $produit;
        return $this;
    }
}